src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -927,16 +949,8 @@ export class BundleDataClient {
                 return;
               }
               // If the fill's repayment address is not a valid EVM address and the repayment chain is an EVM chain, the fill is invalid.
-              if (chainIsEvm(fill.repaymentChainId) && !isValidEvmAddress(fill.relayer)) {
-                const fillTransaction = await originClient.spokePool.provider.getTransaction(fill.transactionHash);
-                const originRelayer = fillTransaction.from;
-                // Repayment chain is still an EVM chain, but the msg.sender is a bytes32 address, so the fill is invalid.
-                if (!isValidEvmAddress(originRelayer)) {
-                  bundleInvalidFillsV3.push(fill);
-                  return;
-                }
-                // Otherwise, assume the relayer to be repaid is the msg.sender.
-                fill.relayer = originRelayer;
+              if (!verifyFill(fill, originClient)) {
```

Comment:
same needs to be await'd

Reply:
https://github.com/across-protocol/sdk/pull/842/commits/a7e77d2a9ac6e47b73b08f1e8c5c56c001b3fe13